extensions/lifecycle/lifecycleConsumer/LifecycleConsumer.js

```diff
@@ -0,0 +1,111 @@
+'use strict'; // eslint-disable-line
+
+const http = require('http');
+const async = require('async');
+const { EventEmitter } = require('events');
+const Logger = require('werelogs').Logger;
+const { errors } = require('arsenal');
+
+const LifecycleObjectTask = require('../tasks/LifecycleObjectTask');
+const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
+
+class ObjectConsumer extends EventEmitter {
+
+    /**
+     * Create a kafka consumer to read lifecycle actions from queue
+     * and perform them on a target S3 endpoint.
+     *
+     * @constructor
+     * @param {Object} zkConfig - zookeeper configuration object
+     * @param {String} zkConfig.connectionString - zookeeper connection string
+     *  as "host:port[/chroot]"
+     * @param {Object} lcConfig - lifecycle configuration object
+     * @param {String} lcConfig.objectTasksTopic - lifecycle object topic name
+     * @param {Object} lcConfig.consumer - kafka consumer object
+     * @param {String} lcConfig.consumer.groupId - kafka consumer group id
+     * @param {Integer} lcConfig.consumer.retryTimeoutS - number of seconds
+     *  before giving up retries of an entry lifecycle action
+     * @param {Integer} lcConfig.consumer.concurrency - number of max allowed
+     *  concurrent operations
+     * @param {Object} s3Config - S3 configuration
+     * @param {Object} s3Config.s3 - s3 endpoint configuration object
+     * @param {Object} authConfig - authentication info on source
+     */
+    constructor(zkConfig, lcConfig, s3Config, authConfig) {
+        super();
+        this.zkConfig = zkConfig;
+        this.lcConfig = lcConfig;
+        this.s3Config = s3Config;
+        this.authConfig = authConfig;
+        this._consumer = null;
+
+        this.logger = new Logger('Backbeat:Lifecycle:ObjectConsumer');
+
+        // global variables
+        // TODO: for SSL support, create HTTPS agents instead
+        this.httpAgent = new http.Agent({ keepAlive: true });
+    }
+
+
+    /**
+     * Start kafka consumer. Emits a 'ready' event when
+     * consumer is ready.
+     *
+     * @return {undefined}
+     */
+    start() {
+        this._consumer = new BackbeatConsumer({
+            zookeeper: {
+                connectionString: this.zkConfig.connectionString,
+            },
+            topic: this.lcConfig.objectTasksTopic,
+            groupId: this.lcConfig.consumer.groupId,
+            concurrency: this.lcConfig.consumer.concurrency,
+            queueProcessor: this.processKafkaEntry.bind(this),
+        });
+        this._consumer.on('error', () => {});
+        this._consumer.subscribe();
+        this.logger.info('queue processor is ready to consume ' +
```

Comment:
The message should be adapted for lifecycle. I recommend:

```
lifecycle consumer successfully started
```

To match the startup message of other lifecycle components.

Reply:
My concern is that we use queue processor for the CRR process, so it may be confused with the "queue processor" of lifecycle.